randovania/games/prime_hunters/exporter/patch_data_factory.py

```diff
@@ -0,0 +1,144 @@
+    return item_id_for_item_resource(game.resource_database.get_item(requirement.item_name))
+
+
+ITEM_SPAWN_ENTITY_TYPE: Final = 4
+ARTIFACT_ENTITY_TYPE: Final = 17
+OCTOLITH_MODEL_ID: Final = 8
+
+_ARTIFACT_TO_MODEL_ID = {
+    "AlinosArtifact1": 0,
+    "AlinosArtifact2": 1,
+    "CelestialArtifact1": 2,
+    "CelestialArtifact2": 3,
+    "VDOArtifact1": 4,
+    "VDOArtifact2": 5,
+    "ArcterraArtifact1": 6,
+    "ArcterraArtifact2": 7,
+}
+
+_OCTOLITH_TO_ARTIFACT_ID = {
+    "Octolith1": 0,
+    "Octolith2": 1,
+    "Octolith3": 2,
+    "Octolith4": 3,
+    "Octolith5": 4,
+    "Octolith6": 5,
+    "Octolith7": 6,
+    "Octolith8": 7,
+}
+
+
+class HuntersPatchDataFactory(PatchDataFactory[HuntersConfiguration, HuntersCosmeticPatches]):
+    def game_enum(self) -> RandovaniaGame:
+        return RandovaniaGame.METROID_PRIME_HUNTERS
+
+    def _calculate_starting_inventory(self, resources: ResourceCollection) -> dict[str, str]:
+        result = {
+            # TODO: Take starting weapons into account
+            "weapons_string": "00000101",
+        }
+        return result
+
+    def _get_pickups_for_area(self, area: Area) -> list:
+        pickups = []
+        pickup_nodes_gen = (node for node in area.nodes if isinstance(node, PickupNode))
+        pickup_nodes = sorted(pickup_nodes_gen, key=lambda n: n.pickup_index)
+
+        for node in pickup_nodes:
+            target = self.patches.pickup_assignment.get(node.pickup_index, None)
+
+            # TODO: Add handling for nothing items
+            assert target is not None
+            pickup: dict = {}
+
+            pickup["entity_id"] = node.extra["entity_id"]
+            entity_type = target.pickup.extra["entity_type"]
+
+            if entity_type == ITEM_SPAWN_ENTITY_TYPE:
+                pickup["entity_type"] = ITEM_SPAWN_ENTITY_TYPE
+                pickup["item_type"] = target.pickup.extra["item_type"]
+            else:
+                pickup["entity_type"] = ARTIFACT_ENTITY_TYPE
+                model_id = target.pickup.extra.get("model_id", None)
+                if model_id == OCTOLITH_MODEL_ID:
+                    pickup["model_id"] = model_id
+                    pickup["artifact_id"] = _OCTOLITH_TO_ARTIFACT_ID[target.pickup.model.name]
+                else:
+                    pickup["model_id"] = _ARTIFACT_TO_MODEL_ID[target.pickup.model.name]
+                    pickup["artifact_id"] = target.pickup.extra["artifact_id"]
+            pickups.append(pickup)
+
+        return pickups
+
+    def _get_force_fields_for_area(self, area: Area, game_specific: dict) -> list:
+        force_fields = []
+
+        for identifier, requirement in game_specific.items():
+            node_identifier = NodeIdentifier.from_string(identifier)
+            if node_identifier.area != area.name:
+                continue
+
+            field = {
+                "entity_id": self.game.region_list.node_by_identifier(NodeIdentifier.from_string(identifier)).extra[
```

Comment:
you can replace the `NodeIdentifier.from_string(identifier)` with just `node_identifier`